_posts/no-orchstration.md

```diff
@@ -0,0 +1,57 @@
+
+![booking service orchestrator diagram](/img/posts/no-orchstration/orchestrator.png){:class="img-fluid mx-auto d-block"}
+
+Let me be clear. The system is alive and kicking. It has been selling tickets for the last few years and didn't have any significant issues. The review was in the context of some concerns the team had with recent changes they were trying to implement.
+
+Without going into the details of the changes, the team was concerned by things like:
+
+-  to add a new payment type, they needed to change the booking service
+- to change the way customers are created/handled, they needed to change the booking service
+- to add groups tickets, they needed to change the booking service
+
+And many more. Every single change they were making to the business process had to go through the booking service. The booking service was a controlling actor in the system. It was acting like a centralizing entity forcing all changes to go through it.
+
+In other words, the booking service violates the single responsibility principle (SRP). It has too many responsibilities. Of course, one could argue that we could design a booking service composed of many classes, each with a single goal. But, even if that's true, it doesn't change the reality that the booking service centralizes all the duties in a single place.
+
+## Why do we think we need a booking service?
+
+We think we need a booking service that orchestrates the process because we fall into the trap of modeling the user mental model. [We model the user information and data representation instead of the behaviors that manipulate those data](https://milestone.topics.it/2021/02/02/do-not-trust-the-user-mental-model.html). Users describe the booking process in such a way that emphasizes the centrality of the booking service. Not to mention that they continuously talk about how information and data are displayed. For example, by dictating that there is the need for a dashboard showing the state of the ongoing business process. ViewModel Composition techniques can solve the [dashboard dilemma](link to composing emails).
+
+## Can we get rid of the orchestrator?
+
+Business processes are not an orchestra; they don't need a director. Business processes are much more like choreography; they need choreographers. Once the organization defines a process, there isn't a need for constant directions. Instead, the process runs, and the organization monitors it instead of meticulously managing it. It's much like a ballet; there is a training phase in which the choreographer instructs the dancers. Then, when the choreography is performed, their role is to monitor and apply tweaks and changes to fix any issue. That is in contrast with an orchestra that requires a director to keep the players in sync.
```

Comment:
```suggestion
Business processes are not an orchestra; they don't need a director. Business processes are much more like choreography; they need choreographers. Once the organization defines a process, there isn't a need for constant directions. Instead, the process runs, and the organization monitors it instead of meticulously managing it. It's much like a ballet; there is a training phase in which the choreographer instructs the dancers. Then, when the choreography is performed, their role is to monitor and apply tweaks and changes to fix any issues. That is in contrast with an orchestra that requires a director to keep the various players in sync.
```